### PR TITLE
Feat: Adding NGC downloading of diagnostic models

### PR DESCRIPTION
**Pull request by NickGeneva**
# Earth-2 MIP Pull Request

## Description

1. Adding the NGC model registry for diagnostic models. Enable users to auto download diagnostics using the correct e2mip:// url. Its built into the diagnostic load_package functions as the default.
Unlike the other models, all diagnostics are stored in one [model registry](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/modulus/models/modulus_diagnostics) making access generally easier.
2. Added a markdown doc because I hate RST... docs need pandoc anyway for converting the Jupytext.
3. Also a few bug fixes: registry download when folder doesnt exist, softmax climatenet output
4. Tested both precip and Climatenet with auto download

Closes: https://github.com/NVIDIA/earth2mip/issues/90

test

test (1)

## Checklist

- [x] I am familiar with the Contributing Guidelines.
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The CHANGELOG.md is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/earth2mip/issues) is linked to this pull request.

### CODE REVIEW

File: docs/diagnostic.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Diagnostics
+
+One way to class AI models in Earth-2 MIP is by those that perform time-integration and those that do not.
```
Comment (yairchn) [inline docs/diagnostic.md:3]: I think "The term diagnostic model refers here to models that do not make a timestep prediction, but simply predict some new channel(s) from existing ones at the same time"

File: docs/diagnostic.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Diagnostic models are a class of models that are intended to transform a set of outputs from a weather/climate model into other quantities of interest.
+Functionally diagnostics are :py:class:`earth2mip.geo_operator.GeoOperator`, models can be viewed as post-processers in the inference process.
+Earth-2 MIP presently provides several built-in diagnostics.
+Those that require a model package files are hosted on [NVIDIA's model registry](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/modulus/models/modulus_diagnostics).
+
+## Motivation
+
+A natural question is: what is the point of developing a diagnostic model over a new weather/climate model that performs time integration?
+Training AI weather models is a very expensive process, requiring a large set of training data and expertise which is not always available.
```
Comment (NickGeneva) [inline docs/diagnostic.md:12]: change to "typically requiring a large set training data, compute and expertise..."

File: docs/diagnostic.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    @classmethod
+    def load_diagnostic(
+        cls, package: Optional[Package], grid: grid.LatLonGrid
+    ):
+        return cls(grid)
+```
+
+## Known Limitations
+
```
Comment (yairchn) [inline docs/diagnostic.md:85]: maybe 
`The diagnostics used must operate on grids and use channels consistent with the weather model it is attached to.`

**Review by yairchn — APPROVED** [review]
LGTM!

File: docs/diagnostic.md (hunk 1 above)
Comment (NickGeneva) [reply docs/diagnostic.md]: Updated! Thanks!

File: docs/diagnostic.md (hunk 3 above)
Comment (NickGeneva) [reply docs/diagnostic.md:86]: Updated!
